nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -612,18 +612,33 @@ private void synchronizeControllerServices(final ProcessGroup group, final Versi
         });
     }
 
-    private void removeMissingConnections(final ProcessGroup group, final VersionedProcessGroup proposed, final Map<String, Connection> connectionsByVersionedId) {
+    private void removeMissingConnections(final ProcessGroup group, final VersionedProcessGroup proposed, final Map<String, Connection> connectionsByVersionedId) { // remove ones with same id but different source as well
```

Comment:
I think the comment here was intended as an internal note to self, perhaps? If so, please remove. It exceeds the 200-character line limit, causing checkstyle failures.